### PR TITLE
Adds C compiler support

### PR DESCRIPTION
**Pull request by ThadHouse**
Requires 1.2.15 of plugin as well.

### CODE REVIEW

File: config.gradle [hunk 1]
```diff
@@ -3,19 +3,22 @@ import org.gradle.internal.os.OperatingSystem
...
 def linuxCompilerArgs = ['-std=c++11', '-Wformat=2', '-Wall', '-Wextra', '-Werror', '-pedantic', '-Wno-psabi', '-g',
                          '-Wno-unused-parameter', '-fPIC', '-rdynamic', '-Wno-error=deprecated-declarations', '-pthread']
+def linuxCCompilerArgs = []
 def linuxLinkerArgs = ['-rdynamic', '-pthread']
 def linuxReleaseCompilerArgs = ['-O2']
 def linuxDebugCompilerArgs = ['-O0']
 def linux32BitArg = '-m32'
 
-def macCompilerArgs =  ['-std=c++11', '-Wall', '-Wextra', '-Werror', '-pedantic-errors', '-fPIC', '-g',
+def maCCompilerArgs =  ['-std=c++11', '-Wall', '-Wextra', '-Werror', '-pedantic-errors', '-fPIC', '-g',
```
Comment (calcmogul) [inline config.gradle]: Missing a lowercase c?

**Comment by ThadHouse** [comment]
Fixed

File: config.gradle [hunk 2]
```diff
@@ -137,6 +143,8 @@ if (!project.hasProperty('onlyAthena')) {
                 architecture = 'x86'
                 operatingSystem = 'linux'
                 compilerArgs = linuxCompilerArgs
+                CCompilerArgs << linux32BitArg
+                CCompilerArgs = linuxCCompilerArgs
                 compilerArgs << linux32BitArg
```
Comment (PeterJohnson) [inline config.gradle]: This is inconsistent ordering compared to the other groups (which put this line immediately after the `compilerArgs` line).

File: config.gradle [hunk 3]
```diff
@@ -3,19 +3,22 @@ import org.gradle.internal.os.OperatingSystem
...
                          '-Wno-unused-parameter', '-fPIC', '-rdynamic', '-Wno-error=deprecated-declarations', '-pthread']
+def linuxCCompilerArgs = []
 def linuxLinkerArgs = ['-rdynamic', '-pthread']
 def linuxReleaseCompilerArgs = ['-O2']
 def linuxDebugCompilerArgs = ['-O0']
 def linux32BitArg = '-m32'
 
 def macCompilerArgs =  ['-std=c++11', '-Wall', '-Wextra', '-Werror', '-pedantic-errors', '-fPIC', '-g',
                             '-Wno-unused-parameter', '-Wno-missing-field-initializers', '-Wno-unused-private-field']
+def macCCompilerArgs = []
```
Comment (PeterJohnson) [inline config.gradle]: I would think the default of these should be similar to the CompilerArgs except for the `'-std=c++11'` part.

**Comment by ThadHouse** [comment]
Fixed the issues. This is now good to go.

File: config.gradle [hunk 4]
```diff
@@ -2,19 +2,24 @@ import edu.wpi.first.nativeutils.*
...
 def windowsCompilerArgs = ['/EHsc', '/DNOMINMAX', '/Zi', '/FS', '/Zc:inline', '/MT']
+def windowsCCompilerArgs = ['/Zi', '/FS', '/Zc:inline', '/MT']
 def windowsReleaseCompilerArgs = ['/O2']
 def windowsLinkerArgs = [ '/DEBUG:FULL' ]
 def windowsReleaseLinkerArgs = [ '/OPT:REF', '/OPT:ICF' ]
 
 def linuxCompilerArgs = ['-std=c++11', '-Wformat=2', '-Wall', '-Wextra', '-Werror', '-pedantic', '-Wno-psabi', '-g',
                          '-Wno-unused-parameter', '-fPIC', '-rdynamic', '-Wno-error=deprecated-declarations', '-pthread']
+def linuxCCompilerArgs = ['-Wformat=2', '-Wall', '-Wextra', '-Werror', '-pedantic', '-Wno-psabi', '-g',
+                         '-Wno-unused-parameter', '-fPIC', '-rdynamic', '-Wno-error=deprecated-declarations', '-pthread']
```
Comment (calcmogul) [inline config.gradle:13]: We don't need `-Wno-error=deprecated-declarations` anymore. master has it removed.

**Review by calcmogul — CHANGES_REQUESTED** [review]

**Pull request closed without merging**